Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -334,55 +315,31 @@ public override List<Cash> GetCashBalance()
+        public List<ExecutionDetails> GetExecutions(string symbol, string type, string exchange, DateTime timeSince, string side)
         {
-            var filter = new IB.ExecutionFilter
+            
+            var filter = new ExecutionFilter()
             {
                 AcctCode = _account,
                 ClientId = _clientID,
                 Exchange = exchange,
-                SecurityType = type ?? IB.SecurityType.Undefined,
+                SecType = type ?? "",
                 Symbol = symbol,
-                Time = timeSince ?? DateTime.MinValue,
-                Side = side ?? IB.ActionSide.Undefined
+                Time = timeSince.ToString("yyyymmdd hh:mm:ss") ?? DateTime.MinValue.ToString("yyyymmdd hh:mm:ss"),
+                Side = side ?? ""
             };
+            
+            var client = new EClientSocket(this);
+            client.eConnect(_host, _port, IncrementClientID());
+            _ibExecutionDetailsResetEvent.Reset();
+            _ibExecutionDetailsRequestId = GetNextRequestID();
+            Client.reqExecutions(_ibExecutionDetailsRequestId, filter);
 
-            var details = new List<IB.ExecDetailsEventArgs>();
-            using (var client = new IB.IBClient())
+            if (!_ibExecutionDetailsResetEvent.WaitOne(5000))
             {
-                client.Connect(_host, _port, IncrementClientID());
-
-                var manualResetEvent = new ManualResetEvent(false);
-
-                int requestID = GetNextRequestID();
-
-                // define our event handlers
-                EventHandler<IB.ExecutionDataEndEventArgs> clientOnExecutionDataEnd = (sender, args) =>
-                {
-                    if (args.RequestId == requestID) manualResetEvent.Set();
-                };
-                EventHandler<IB.ExecDetailsEventArgs> clientOnExecDetails = (sender, args) =>
-                {
-                    if (args.RequestId == requestID) details.Add(args);
-                };
-
-                client.ExecDetails += clientOnExecDetails;
-                client.ExecutionDataEnd += clientOnExecutionDataEnd;
-
-                // no need to be fancy with request id since that's all this client does is 1 request
-                client.RequestExecutions(requestID, filter);
-
-                if (!manualResetEvent.WaitOne(5000))
-                {
-                    throw new TimeoutException("InteractiveBrokersBrokerage.GetExecutions(): Operation took longer than 1 second.");
-                }
-
-                // remove our event handlers
-                client.ExecDetails -= clientOnExecDetails;
-                client.ExecutionDataEnd -= clientOnExecutionDataEnd;
+                throw new TimeoutException("InteractiveBrokersBrokerage.GetExecutions(): Operation took longer than 1 second.");
             }
-
-            return details;
+            return _exectionDetailsList;
```

Comment:
I have created one more property for the `ExecutionDetails` class for the reset event. Since the key is the request id, a reset event will be created per request id. The event will be "Set()" once it hits the `execDetailsEnd` gets a response indicating that the Executions have arrived for a particular request id.
Take a look at my latest commit.
Let me know if the implementation seems correct.

the other way could be to create the `ConcurrentDictionary` something like:
`ConcurrentDictionary<int, Tuple<ExecutionDetails, ManualResetEvent>>`